lib/puppet/type/file/ensure.rb

```diff
@@ -113,7 +113,10 @@ module Puppet
 
     munge do |value|
       value = super(value)
-      value, resource[:target] = :link, value unless value.is_a? Symbol
+      unless value.is_a? Symbol
+        resource[:target] = value
```

Comment:
I think the order of these two should be reversed, like

```
value = :link
resource[:target] = value
```

Since that was the order they were assigned in the original expression

Reply:
This makes sense to me but weirdly enough reversing the order causes `./spec/unit/type/file_spec.rb:1276` to fail

Reply:
Oh, I think I see. Those two variables were assigned in parallel before, so `resource[:target]` was being assigned the `value` set in 115, not the `value` set to `:link`. Weird!